heidrun/mdl_map.rb

```diff
@@ -0,0 +1,77 @@
+Krikri::Mapper.define(:mdl_map, :parser => Krikri::JsonParser) do
+  provider :class => DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/mdl'
+    label record.field('record', 'provider')
+  end
+
+  dataProvider :class => DPLA::MAP::Agent do
+    providedLabel record.field('record', 'dataProvider')
+  end
+
+  isShownAt :class => DPLA::MAP::WebResource do
+    uri record.field('record', 'isShownAt')
+  end
+
+  preview :class => DPLA::MAP::WebResource do
+    uri record.field('record' 'object')
+  end
+
+  originalRecord :class => DPLA::MAP::WebResource do
+    uri record_uri
+  end
+
+  sourceResource :class => DPLA::MAP::SourceResource do
+  
+    collection :class => DPLA::MAP::Collection, :each => header.field('record', 'sourceResource', 'collection'), :as => :coll do
+      title coll.field('title')
+      description coll.field('description', 'dc', 'description')
+    end
+    #need to verify this mapping with MDL
+
+    contributor :class => DPLA::MAP::Agent, :each=> record.field('record', 'sourceResource', 'contributor'), :as => :contributor do
+      providedLabel contributor
+    end
+    
+    creator :class => DPLA::MAP::Agent, :each => record.field('record', 'sourceResource', 'creator'), :as => :creator do
+      providedLabel creator
+    end
+
+    date :class => DPLA::MAP::TimeSpan, :each => record.field('record', 'sourceResource', 'date', 'displayDate'), :as => :created do
+      providedLabel created
+      self.begin created.field('record', 'sourceResource', 'date', 'begin')
+      self.end created.field('record', 'sourceResource', 'date', 'end')
+    end
+
+    description record.field('record', 'sourceResource', 'description')
+
+    dcformat record.field('record', 'sourceResource', 'format')
```

Comment:
We actually didn't mention that to them. I'm not sure I knew that was an issue. A cursory look at the documents in their api shows them all as children of sourceResource, so I think we should be good with this as is, and I can include this as part of QA to explore further.